src/lightmap.cpp

```diff
@@ -254,30 +254,58 @@ void map::build_sunlight_cache( int zlev )
     const auto &prev_floor_cache = prev_map_cache.floor_cache;
     const auto &outside_cache = map_cache.outside_cache;
     const float sight_penalty = get_weather().weather_id->sight_penalty;
-    for( int x = 0, prev_x = offset.x; x < MAPSIZE_X; x++, prev_x++ ) {
-        bool x_inbounds = prev_x >= 0 && prev_x < MAPSIZE_X;
-        for( int y = 0, prev_y = offset.y; y < MAPSIZE_Y; y++, prev_y++ ) {
-            bool inbounds = x_inbounds && prev_y >= 0 && prev_y < MAPSIZE_Y;
-            four_quadrants prev_light( outside_light_level );
-            float prev_transparency = static_cast<float>( LIGHT_TRANSPARENCY_OPEN_AIR );
-            if( inbounds ) {
-                prev_light = prev_lm[ prev_x ][ prev_y ];
-                prev_transparency = prev_transparency_cache[ prev_x ][ prev_y ];
-                // This is pretty gross, this cancels out the per-tile transparency effect
-                // derived from weather.
-                if( outside_cache[x][y] ) {
-                    prev_transparency /= sight_penalty;
+    // These are N, E, S, W.
```

Comment:
This comment seems wrong?  Should be N, W, E, S (but now you're using the named constants it can probably just be deleted).